sentinel/transitions/accept_patient_reports.js

```diff
@@ -81,16 +81,25 @@ const getConfig = function(form) {
     return _.findWhere(fullConfig, { form: form });
 };
 
-const _silenceReminders = (audit, registration, reported_date, config, callback) => {
-    var toClear = module.exports._findToClear(registration, reported_date, config);
+const _silenceReminders = (audit, registration, report, config, callback) => {
+    var toClear = module.exports._findToClear(registration, report.reported_date, config);
     if (!toClear.length) {
         return callback();
     }
 
-    toClear.forEach(task => utils.setTaskState(task, 'cleared'));
+    toClear.forEach(task => {
+        utils.setTaskState(task, 'cleared');
+        task.cleared_by = report._id;
+    });
     audit.saveDoc(registration, callback);
 };
 
+const addRegistrationToDoc = (doc, registrations) => {
+    if (registrations.length) {
+        doc.registration_id = registrations[registrations.length - 1]._id;
```

Comment:
Have you checked that registration order is useful? I'm not sure that the last one in the list is necessarily the most recent or the most useful. The view just emits by patientId, so I think it's effectively ordered arbitrarily?

If it is actually arbitrary I worry that we're codifying incorrect knowledge here: that someone will think in the future that there is a logical order to registrations returned when there isn't. Then they find out that it's not in the view, then they think it's a bug and change the view… etc.

Reply:
Good point. I should pick the latest by reported_date.